modules/fundamental/src/purl/service/mod.rs

```diff
@@ -233,47 +232,40 @@ impl PurlService {
     }
 
     #[instrument(skip(self, connection), err(level=tracing::Level::INFO))]
-    pub async fn fetch_purl_details<C: ConnectionTrait>(
+    pub async fn fetch_purl_details<C: ConnectionTrait, I: AsRef<str> + Debug>(
         &self,
-        identifiers: &[&str],
+        identifiers: &[I],
         deprecated: Deprecation,
         connection: &C,
-    ) -> Result<HashMap<String, PurlDetails>, String> {
```

Comment:
Pretty much every module has an `Error` struct that does convenient macro-y things. Rarely (if at all) within trustify are our Results returning some type of `Error`, not a `String`.